Documentation/Books/Manual/Programs/Arangodump/Maskings.md

```diff
@@ -88,14 +109,140 @@ or
+If the attribute value is an array the masking is applied to all the
+array elements.
+
+### Example
+
+This following will replace name by a "XXXX" masked string
+
+    {
+      "type": "xify_front",
+      "path": ".name",
+      "unmaskedLength": 2
+    }
+
+The object
+
+    {
+      "name": "top-level-name",
+      "age": 42,
+      "nicknames" : [ { "name": "hugo" }, "egon" ],
+      "other": {
+        "name": [ "emil", { "secret": "superman" } ]
+      }
+    }
+
+will be replaced as follows:
+
+    {
+      "name": "xxxxxxxxxxxxme",
+      "age": 42,
+      "nicknames" : [ { "name": "xxgo" }, "egon" ],
+      "other": {
+        "name": [ "xxil", { "secret": "superman" } ]
+      }
+      }
+
+`egon` and `superman` are not replaced, because they are not in an
+attribute value of an attribute `name`.
+
+### sub-objects and arrays
+
+If you specify a path and the attribute value is an array then the
+masking decision is applied to each entry of this array as if this
+were the value of the attribute.
+
+If the attribute value is a object, then the attribute is not masked.
+Instead the sub-object is checked further for leaf attributes.
+
+#### Example
+
+Masking `email` will convert
 
-This masking replaces characters with `x` and ` `. Alphanumeric characters,
-`_` and `-` are replaced by `x`, everything else is replaced by ` `.
+
+  { 
+    "email" : "email address" 
+  }
+
+into
+
+  { 
+    "email" : "xxil xxxxxxss" 
+  }
+
+because `email` is a leaf sttribute.
+
+  { 
+    "email" : [ 
+      "address one", 
+      "address two" 
+    ] 
+  } 
+
+will be converted into
+
+  { 
+    "email" : [ 
+      "xxxxxss xne", 
+      "xxxxxss xwo" 
+    ] 
+  } 
+
+because the array is "unfolded".
+
+  { 
+    "email" : { 
+      "address" : "email address" 
+    } 
+    }
+
+will not be changed because `email` is not a leaf attribute.
+
+
+Masking Functions
+-----------------
+
+### random_string
+
+This masking replaces character strings with a random string of the
+same length. Other values, e.g. numbers and booleans, are not changed.
 
     {
       "path": ".name",
+      "type": "random_string"
+    }
+
+This will replace all values of attributes with key `name` with a
+anonymized string. It is not guaranteed that the string will be of
+the same length.
+
+A hash of the original string is computed. If the original string is
+shorted then the hash will be used. This will result in a longer
+string. If the string is longer than hash than the hash characters
+will be repeated as many times as needed to reach the full original
+string length.
+
+The random string will use the characters `a` to `z` and `A`to `Z`.
+
+    {
+      "path": ".name",
+      "type": "random_string",
+      "characterSet": "abcd01234"
+    }
+
+In this case the random string will use the characters `a`, `b`, `c`,
+`d` as well as the digits `0`, `1`, `2`, `3`, `4`. If `characterSet`
+is empty, it defaults to `a` to `z` and `A` to `Z`.
+
+### xify_front (Enterprise)
```

Comment:
Is _xify_front_ a common name for this?

Is this meant for enterprises, or is it only available in the Enterprise Edition?

Reply:
done